pkg/operator/reports_test.go

```diff
@@ -92,3 +101,177 @@ func TestGetNextReportPeriod(t *testing.T) {
+	reportEnd := &reportEndTmp
+
+	testTable := []struct {
+		name          string
+		report        *metering.Report
+		expectRunning bool
+	}{
+		{
+			name:          "new report returns false",
+			report:        testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, false),
+			expectRunning: false,
+		},
+		{
+			name: "finished status on run-once report returns true",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+			}, nil, false),
+			expectRunning: true,
+		},
+		{
+			name: "reportingEnd is unset returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, nil, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectRunning: false,
+		},
+		{
+			name: "reportingEnd > lastReportTime returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+				LastReportTime: &metav1.Time{Time: reportStart.AddDate(0, 0, 0)},
+			}, schedule, false),
+			expectRunning: false,
+		},
+		{
+			name: "reportingEnd < lastReportTime returns true",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+				LastReportTime: &metav1.Time{Time: reportStart.AddDate(0, 2, 0)},
+			}, schedule, false),
+			expectRunning: true,
+		},
+		{
+			name: "spec.Reason set to 'Scheduled' returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ScheduledReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectRunning: false,
+		},
+		{
+			name: "spec.Reason set to 'InvalidReport' returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.InvalidReportReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectRunning: false,
+		},
+	}
```

Comment:
We need at least 1 case where running is true (`v1.ConditionTrue`), maybe more.

Reply:
Added - wasn't able to think of any other paths that weren't tested.

Reply:
I guess our test coverage files would probably help with that. You can verify by running `make test` and it will output `coverage.out` which can be used with the `go tool cover` command: https://blog.golang.org/cover, want to do a check for us?